core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/If.java

```diff
@@ -0,0 +1,190 @@
+    public Object getInput() {
+        return input;
+    }
+
+    @Override
+    public void setInput(final Object input) {
+        this.input = input;
+    }
+
+    @Override
+    public TypeReference<Object> getOutputTypeReference() {
+        return new TypeReferenceImpl.Object();
+    }
+
+    @Override
+    public If shallowClone() throws CloneFailedException {
+        return new If.Builder()
+                .input(input)
+                .condition(condition)
+                .predicate(predicate)
+                .then(then)
+                .otherwise(otherwise)
+                .build();
+    }
+
+    @Override
+    public Map<String, String> getOptions() {
+        return options;
+    }
+
+    @Override
+    public void setOptions(final Map<String, String> options) {
+        this.options = options;
+    }
+
+    @Override
+    public Collection getOperations() {
+        return null;
+    }
+
+    public Boolean getCondition() {
+        return condition;
+    }
+
+    public void setCondition(final boolean condition) {
+        this.condition = condition;
+    }
+
+    public Operation getThen() {
+        return then;
+    }
+
+    public void setThen(final Operation then) {
+        this.then = then;
+    }
+
+    public Operation getOtherwise() {
+        return otherwise;
+    }
+
+    public void setOtherwise(final Operation otherwise) {
+        this.otherwise = otherwise;
+    }
+
+    public Predicate<Object> getPredicate() {
+        return predicate;
+    }
+
+    public void setPredicate(final Predicate<Object> predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (null == obj || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        final If filter = (If) obj;
+
+        return new EqualsBuilder()
+                .append(condition, filter.condition)
+                .append(predicate, filter.predicate)
+                .append(then, filter.then)
+                .append(otherwise, filter.otherwise)
+                .isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(31, 83)
+                .append(condition)
+                .append(predicate)
+                .append(then)
+                .append(otherwise)
+                .toHashCode();
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .append(condition)
+                .append(predicate)
+                .append(then)
+                .append(otherwise)
+                .toString();
+    }
+
+    public static final class Builder extends Operation.BaseBuilder<If, Builder>
+        implements InputOutput.Builder<If, Object, Object, Builder> {
+        public Builder() {
+            super(new If());
+        }
+
+        public Builder condition(final boolean condition) {
```

Comment:
I think you'll need to change the parameter to a Boolean, otherwise your shallowClone method will fail if 'condition' is not set.